framework/src/main/java/org/checkerframework/common/accumulation/AccumulationChecker.java

```diff
@@ -0,0 +1,30 @@
+package org.checkerframework.common.accumulation;
+
+import java.util.LinkedHashSet;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.returnsreceiver.ReturnsReceiverChecker;
+
+/**
+ * An accumulation checker is one that accumulates some property: method calls, keys into maps, etc.
+ * All accumulation analyses share some common features: their type systems are similar in
+ * structure, they need to reason about fluent APIs, and they need some way to easily add facts to
+ * the accumulation they are tracking. This accumulation analysis represents all facts as Strings.
+ *
+ * <p>Accumulation checkers are particularly interesting because they can represent many
+ * typestate-like properties, but do not require a precise alias analysis for soundness.
+ *
+ * <p>This class provides a basic accumulation checker that can be extended to implement a
+ * particular accumulation type system. The primary extension point is the constructor of {@link
+ * AccumulationAnnotatedTypeFactory}, which every subclass should override to provide custom
+ * annotations.
+ */
+public abstract class AccumulationChecker extends BaseTypeChecker {
+
+    @Override
+    protected LinkedHashSet<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
+        LinkedHashSet<Class<? extends BaseTypeChecker>> checkers =
+                super.getImmediateSubcheckerClasses();
+        checkers.add(ReturnsReceiverChecker.class);
```

Comment:
Is it possible someone might want to build an accumulation checker that does not require returns-receiver reasoning?  Or shall we just assume any accumulation checker is also dealing with fluent APIs?

Reply:
I assumed that anyone building an accumulation checker would want RR reasoning - there's not much reason to *not* include it. I suppose we could make it configurable, somehow (but then again, it already is - someone who didn't want the RR checker could override this method again!).

I'll add something to the javadoc that mentions it.

Reply:
There _is_ the whole including a sub-checker doubles the overall checking time issue... 🙂But this is fine for now